core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -214,8 +287,13 @@ private WriterWrapper getWriter(Object key) {
 
   @Override
   protected void cleanUncommitted(Set<ManifestFile> committed) {
-    for (ManifestFile manifest : newManifests) {
-      if (!committed.contains(manifest)) {
+    cleanUncommitted(newManifests, committed);
+    cleanUncommitted(addedManifests, committed);
```

Comment:
Should this clean up added manifest files?

I think that is a responsibility of the caller because the caller may be able to reuse some of them if the commit fails because a deleted manifest is no longer in the table. This is similar to a write operation, where if Iceberg can't commit the write, it doesn't delete the data files that were added in the failed operation. Instead, the writer has abort functions that are called outside of the snapshot commit: https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/Writer.java#L183

Reply:
This cleans up copies of the added manifests, not the original manifests that were added. I believe the caller doesn't have access to the copied manifests.

Reply:
Ah, you're right. Thanks for the explanation!